File: Go-Solutions/offer/21.go
package offer

func exchange(nums []int) []int {
	if nums == nil || len(nums) == 0 {
		return nums
	}
	n := len(nums)
	result := make([]int, n)
	left := 0
	right := n - 1
	for i := 0; i < n; i++ {
		if nums[i] % 2 == 1 {
			result[left] = nums[i]
			left++
		} else {
			result[right] = nums[i]
			right--
		}
	}
	return result
}

File: Java-Kotlin/src/main/java/offer/_21.java
package offer;

public class _21 {

    public int[] exchange(int[] nums) {
        if (nums == null || nums.length == 0) {
            return nums;
        }
        int[] result = new int[nums.length];
        int i = 0;
        int j = result.length - 1;
        for (int num : nums) {
            if (num % 2 == 1) {
                result[i] = num;
                i++;
            } else {
                result[j] = num;
                j--;
            }
        }
        return result;
    }
}
